pyesbulk/__init__.py

```diff
@@ -113,31 +217,41 @@ def put_template(es, name=None, mapping_name=None, body=None):
         Failure modes are raised as exceptions.
     """
     assert name is not None and mapping_name is not None and body is not None
+    _import_elasticsearch(es)
+    logger = client_logger
+
+    # Piggyback on the Elasticsearch logger if the client doesn't give one.
+    if not logger:
+        logger = logging.getLogger("elasticsearch")
     retry = True
     retry_count = 0
     original_no_version = False
     backoff = 1
     beg, end = time.time(), None
-    try:
-        body_ver = int(body["mappings"][mapping_name]["_meta"]["version"])
-    except KeyError:
-        raise Exception(
-            f"Bad template, {name}: mapping name, '{mapping_name}',"
-            " missing from template"
-        )
+    mapping = body["mappings"]
+    body_ver = _get_meta_version(name, mapping)
     while retry:
```

Comment:
Would the use of `global es_excs, helpers` here to avoid the use `noqa` for flake8 below?